File: examples/python/testing-patterns/complex-workflows/test-maximum-attempts.py
@pytest.mark.durable_execution(handler=handler, lambda_function_name="max_attempts")
def test_max_attempts(durable_runner):
    """Test polling stops at max attempts."""
    with durable_runner:
        result = durable_runner.run(input={"target": 10}, timeout=30)
    
    assert result.status is InvocationStatus.SUCCEEDED
    
    final_result = deserialize_operation_payload(result.result)
    assert final_result["attempts"] == 5
    assert final_result["state"] == 5
    assert final_result["reached_target"] is False

File: examples/java/testing-patterns/complex-workflows/test-maximum-attempts.java
@Test
void testMaxAttempts() {
    var runner = LocalDurableTestRunner.create(Map.class, new MaxAttemptsHandler());

    var result = runner.runUntilComplete(Map.of("target", 10));

    assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());

    var finalResult = result.getResult(Map.class);
    assertEquals(5, finalResult.get("attempts"));
    assertEquals(5, finalResult.get("state"));
    assertEquals(false, finalResult.get("reached_target"));
}
